spec/controllers/people_controller_spec.rb

```diff
@@ -0,0 +1,117 @@
+# spec/controllers/people_controller_spec.rb
+require 'rails_helper'
+
+RSpec.describe PeopleController, type: :controller do
+  describe "GET #index" do
+    subject { get :index }
+
+    it "renders the template" do
+      expect(subject).to render_template(:index)
+    end
+    it "loads all of the people into @people" do
+      expect(subject.body.length).to be_truthy
+    end
+  end
+
+  describe "GET #show" do
+    subject { get :show, params: { id: person } }
+    let(:person) { FactoryGirl.create :person }
+
+    it "renders the template" do
+      expect(subject).to render_template(:show)
+    end
+    it "loads the person into @person" do
+      expect(assigns(person)).to eq(@person)
+    end
+  end
+
+  describe "GET #new" do
+    subject { get :new }
+
+    it "renders the template" do
+      expect(subject).to render_template :new 
+    end
+  end
+
+  describe "GET #edit" do
+    subject { get :edit, params: { id: person.id } }
+    let(:person) { FactoryGirl.create :person }
+
+    it "renders the template" do
+      expect(subject).to render_template :edit
+    end
+    it "loads the person into @person" do
+      expect(assigns(person)).to eq(@person)
+    end
+  end
+
+  describe "POST #create" do
+    subject { post :create, params: { person: params } }
+    let(:params) { FactoryGirl.attributes_for(:person) }
+
+    it "changes person count by +1" do
+      expect { subject }.to change(Person, :count).by(1)
+    end
+
+    it "loads the person into @person" do
+      expect(subject).to redirect_to(Person.last)
+    end
+
+    context "with invalid attributes" do
+      let(:params) { FactoryGirl.attributes_for(:person, first_name: nil) }
+
+      it "does not save the new person" do
+        expect{ subject }.to_not change(Person, :count)
+      end
+
+      it "re-runs people#new" do
+        expect(subject).to render_template :new
+      end
+    end
+  end
+
+  describe "PATCH #update" do
+    let!(:person) { FactoryGirl.create(:person) }
+    subject { put :update, params: { id: person, person: params } }
+
+    it "should located requested person" do
+      expect(assigns(person)).to eq(@person)
+    end
+
+    context "changing name" do 
+      let(:params) { { first_name: 'Jane' } }
+      it "should update person" do
+        subject
+        person.reload
+        expect(person.first_name).to eq('Jane')
```

Comment:
One great way we could write this test to be super explicit about what we're testing would be:

```ruby
it "updates the person's name" do
  expect { subject }.to change { person.first_name }.from("Jessica").to("Jane")
end
```